### PR TITLE
[FAB-17422] Add source component field to metrics

### PR DESCRIPTION
**Pull request by stephyee**
Signed-off-by: Tiffany Harris <tiffany.harris@ibm.com>

#### Type of change

- Documentation update

#### Description

Add source component field to metrics, including documentation

#### Related issues
[FAB-17422](https://jira.hyperledger.org/browse/FAB-17422)

### CODE REVIEW

File: docs/source/metrics_reference.rst [hunk 1]
```diff
@@ -6,324 +6,324 @@ Prometheus Metrics
...
-| ledger_transaction_count                            | counter   | Number of transactions processed.                          | channel          |                                                             |
-|                                                     |           |                                                            +------------------+-------------------------------------------------------------+
-|                                                     |           |                                                            | transaction_type |                                                             |
-|                                                     |           |                                                            +------------------+-------------------------------------------------------------+
-|                                                     |           |                                                            | chaincode        |                                                             |
-|
```
Comment (pamandrejko) [inline docs/source/metrics_reference.rst]: If I am reading this correctly, it looks like some of the metrics may not have a Source? Is that intentional?

**Review by pamandrejko — COMMENTED** [review]
Thank you Tiffany for adding this. This is great.

**Review by mastersingh24 — DISMISSED** [review]

File: core/chaincode/metrics.go [hunk 2]
```diff
@@ -10,54 +10,61 @@ import "github.com/hyperledger/fabric/common/metrics"
...
-		Name:         "launch_duration",
-		Help:         "The time to launch a chaincode.",
-		LabelNames:   []string{"chaincode", "success"},
-		StatsdFormat: "%{#fqname}.%{chaincode}.%{success}",
+		Namespace:       "chaincode",
+		Name:            "launch_duration",
+		Help:            "The time to launch a chaincode.",
+		LabelNames:      []string{"chaincode", "success"},
+		StatsdFormat:    "%{#fqname}.%{chaincode}.%{success}",
+		SourceComponent: "chaincode",
```
Comment (sykesm) [inline core/chaincode/metrics.go]: Here we have a metric that isn't "peer" or "orderer" as documented where `SourceComponent` was defined.

It's also confusing as this is emitted by the peer for chaincode that it has launched.

File: docs/source/metrics_reference.rst (hunk 1 above)
Comment (sykesm) [reply docs/source/metrics_reference.rst]: "Source" is a misnomer and part of why the request is not really trivial. Metrics are in namespaces and are simply emitted. Some of these emissions come from the peer, some from the order, some from both. There are others that are specific to the CA and not documented here.

Comment (sykesm) [inline docs/source/metrics_reference.rst:431]: I'd have expected the StatsD metrics to include similar documentation.

**Review by sykesm — CHANGES_REQUESTED** [review]
- The naming inconsistency is not a big deal but it's confusing. It would be better if it weren't.
- The doc updates should apply to both statsd and prometheus.

As for overall effort, I question whether adding the grouping to each metric is the right way to handle this. An alternative would be to provide a map of namespace to some _grouping_ that is documented. This would be contained within gendoc and the script we use to generate it.

At the end of the day, metrics are emitted. In the case of prometheus, the source is the thing being scraped and that information is captured with the metric. In the case of statsd, the metrics.statsd.prefix is prepended as a source identifier.

The thing that Pam wants help "solving" is how to document what metrics are available on a peer, an orderer, and the CA. The CA is relatively simple since it only has a single server process and includes its own documentation. The peer and orderer are more difficult because they share a lot of code.

So, to make things relatively accurate in an automated way, I think we can adapt most of what we have fairly quickly. Here's a snippet that just captures the orderer metrics:

```sh
orderer_deps=()
while IFS= read -r pkg; do orderer_deps+=("$pkg"); done < <(go list -deps github.com/hyperledger/fabric/cmd/orderer | sort -u | grep hyperledger)

gendoc_command="go run github.com/hyperledger/fabric/common/metrics/cmd/gendoc -template ${metrics_template} ${orderer_deps[*]}"
```

Results in
```
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| Bucket                                                                    | Type      | Description                                                |
+===========================================================================+===========+============================================================+
| blockcutter.block_fill_duration.%{channel}                                | histogram | The time from first transaction enqueing to the block      |
|                                                                           |           | being cut in seconds.                                      |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| broadcast.enqueue_duration.%{channel}.%{type}.%{status}                   | histogram | The time to enqueue a transaction in seconds.              |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| broadcast.processed_count.%{channel}.%{type}.%{status}                    | counter   | The number of transactions processed.                      |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| broadcast.validate_duration.%{channel}.%{type}.%{status}                  | histogram | The time to validate a transaction in seconds.             |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| cluster.comm.egress_queue_capacity.%{host}.%{msg_type}.%{channel}         | gauge     | Capacity of the egress queue.                              |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| cluster.comm.egress_queue_length.%{host}.%{msg_type}.%{channel}           | gauge     | Length of the egress queue.                                |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| cluster.comm.egress_queue_workers.%{channel}                              | gauge     | Count of egress queue workers.                             |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| cluster.comm.egress_stream_count.%{channel}                               | gauge     | Count of streams to other nodes.                           |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| cluster.comm.egress_tls_connection_count                                  | gauge     | Count of TLS connections to other nodes.                   |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| cluster.comm.ingress_stream_count                                         | gauge     | Count of streams from other nodes.                         |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| cluster.comm.msg_dropped_count.%{host}.%{channel}                         | counter   | Count of messages dropped.                                 |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| cluster.comm.msg_send_time.%{host}.%{channel}                             | histogram | The time it takes to send a message in seconds.            |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| consensus.etcdraft.active_nodes.%{channel}                                | gauge     | Number of active nodes in this channel.                    |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| consensus.etcdraft.cluster_size.%{channel}                                | gauge     | Number of nodes in this channel.                           |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| consensus.etcdraft.committed_block_number.%{channel}                      | gauge     | The block number of the latest block committed.            |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| consensus.etcdraft.config_proposals_received.%{channel}                   | counter   | The total number of proposals received for config type     |
|                                                                           |           | transactions.                                              |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| consensus.etcdraft.data_persist_duration.%{channel}                       | histogram | The time taken for etcd/raft data to be persisted in       |
|                                                                           |           | storage (in seconds).                                      |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| consensus.etcdraft.is_leader.%{channel}                                   | gauge     | The leadership status of the current node: 1 if it is the  |
|                                                                           |           | leader else 0.                                             |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| consensus.etcdraft.leader_changes.%{channel}                              | counter   | The number of leader changes since process start.          |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| consensus.etcdraft.normal_proposals_received.%{channel}                   | counter   | The total number of proposals received for normal type     |
|                                                                           |           | transactions.                                              |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| consensus.etcdraft.proposal_failures.%{channel}                           | counter   | The number of proposal failures.                           |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| consensus.etcdraft.snapshot_block_number.%{channel}                       | gauge     | The block number of the latest snapshot.                   |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| consensus.kafka.batch_size.%{topic}                                       | gauge     | The mean batch size in bytes sent to topics.               |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| consensus.kafka.compression_ratio.%{topic}                                | gauge     | The mean compression ratio (as percentage) for topics.     |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| consensus.kafka.incoming_byte_rate.%{broker_id}                           | gauge     | Bytes/second read off brokers.                             |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| consensus.kafka.last_offset_persisted.%{channel}                          | gauge     | The offset specified in the block metadata of the most     |
|                                                                           |           | recently committed block.                                  |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| consensus.kafka.outgoing_byte_rate.%{broker_id}                           | gauge     | Bytes/second written to brokers.                           |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| consensus.kafka.record_send_rate.%{topic}                                 | gauge     | The number of records per second sent to topics.           |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| consensus.kafka.records_per_request.%{topic}                              | gauge     | The mean number of records sent per request to topics.     |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| consensus.kafka.request_latency.%{broker_id}                              | gauge     | The mean request latency in ms to brokers.                 |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| consensus.kafka.request_rate.%{broker_id}                                 | gauge     | Requests/second sent to brokers.                           |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| consensus.kafka.request_size.%{broker_id}                                 | gauge     | The mean request size in bytes to brokers.                 |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| consensus.kafka.response_rate.%{broker_id}                                | gauge     | Requests/second sent to brokers.                           |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| consensus.kafka.response_size.%{broker_id}                                | gauge     | The mean response size in bytes from brokers.              |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| couchdb.processing_time.%{database}.%{function_name}.%{result}            | histogram | Time taken in seconds for the function to complete request |
|                                                                           |           | to CouchDB                                                 |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| deliver.blocks_sent.%{channel}.%{filtered}.%{data_type}                   | counter   | The number of blocks sent by the deliver service.          |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| deliver.requests_completed.%{channel}.%{filtered}.%{data_type}.%{success} | counter   | The number of deliver requests that have been completed.   |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| deliver.requests_received.%{channel}.%{filtered}.%{data_type}             | counter   | The number of deliver requests that have been received.    |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| deliver.streams_closed                                                    | counter   | The number of GRPC streams that have been closed for the   |
|                                                                           |           | deliver service.                                           |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| deliver.streams_opened                                                    | counter   | The number of GRPC streams that have been opened for the   |
|                                                                           |           | deliver service.                                           |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| fabric_version.%{version}                                                 | gauge     | The active version of Fabric.                              |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| grpc.comm.conn_closed                                                     | counter   | gRPC connections closed. Open minus closed is the active   |
|                                                                           |           | number of connections.                                     |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| grpc.comm.conn_opened                                                     | counter   | gRPC connections opened. Open minus closed is the active   |
|                                                                           |           | number of connections.                                     |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| grpc.server.stream_messages_received.%{service}.%{method}                 | counter   | The number of stream messages received.                    |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| grpc.server.stream_messages_sent.%{service}.%{method}                     | counter   | The number of stream messages sent.                        |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| grpc.server.stream_request_duration.%{service}.%{method}.%{code}          | histogram | The time to complete a stream request.                     |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| grpc.server.stream_requests_completed.%{service}.%{method}.%{code}        | counter   | The number of stream requests completed.                   |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| grpc.server.stream_requests_received.%{service}.%{method}                 | counter   | The number of stream requests received.                    |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| grpc.server.unary_request_duration.%{service}.%{method}.%{code}           | histogram | The time to complete a unary request.                      |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| grpc.server.unary_requests_completed.%{service}.%{method}.%{code}         | counter   | The number of unary requests completed.                    |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| grpc.server.unary_requests_received.%{service}.%{method}                  | counter   | The number of unary requests received.                     |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| ledger.blockchain_height.%{channel}                                       | gauge     | Height of the chain in blocks.                             |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| ledger.blockstorage_commit_time.%{channel}                                | histogram | Time taken in seconds for committing the block to storage. |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| logging.entries_checked.%{level}                                          | counter   | Number of log entries checked against the active logging   |
|                                                                           |           | level                                                      |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| logging.entries_written.%{level}                                          | counter   | Number of log entries that are written                     |
+---------------------------------------------------------------------------+-----------+------------------------------------------------------------+

```

And the peer:
```sh
peer_deps=()
while IFS= read -r pkg; do peer_deps+=("$pkg"); done < <(go list -deps github.com/hyperledger/fabric/cmd/peer | sort -u | grep hyperledger/fabric/)

gendoc_command="go run github.com/hyperledger/fabric/common/metrics/cmd/gendoc -template ${metrics_template} ${peer_deps[*]}"
```

Results in
```
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| Bucket                                                                                  | Type      | Description                                                |
+=========================================================================================+===========+============================================================+
| chaincode.execute_timeouts.%{chaincode}                                                 | counter   | The number of chaincode executions (Init or Invoke) that   |
|                                                                                         |           | have timed out.                                            |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| chaincode.launch_duration.%{chaincode}.%{success}                                       | histogram | The time to launch a chaincode.                            |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| chaincode.launch_failures.%{chaincode}                                                  | counter   | The number of chaincode launches that have failed.         |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| chaincode.launch_timeouts.%{chaincode}                                                  | counter   | The number of chaincode launches that have timed out.      |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| chaincode.shim_request_duration.%{type}.%{channel}.%{chaincode}.%{success}              | histogram | The time to complete chaincode shim requests.              |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| chaincode.shim_requests_completed.%{type}.%{channel}.%{chaincode}.%{success}            | counter   | The number of chaincode shim requests completed.           |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| chaincode.shim_requests_received.%{type}.%{channel}.%{chaincode}                        | counter   | The number of chaincode shim requests received.            |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| couchdb.processing_time.%{database}.%{function_name}.%{result}                          | histogram | Time taken in seconds for the function to complete request |
|                                                                                         |           | to CouchDB                                                 |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| deliver.blocks_sent.%{channel}.%{filtered}.%{data_type}                                 | counter   | The number of blocks sent by the deliver service.          |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| deliver.requests_completed.%{channel}.%{filtered}.%{data_type}.%{success}               | counter   | The number of deliver requests that have been completed.   |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| deliver.requests_received.%{channel}.%{filtered}.%{data_type}                           | counter   | The number of deliver requests that have been received.    |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| deliver.streams_closed                                                                  | counter   | The number of GRPC streams that have been closed for the   |
|                                                                                         |           | deliver service.                                           |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| deliver.streams_opened                                                                  | counter   | The number of GRPC streams that have been opened for the   |
|                                                                                         |           | deliver service.                                           |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| dockercontroller.chaincode_container_build_duration.%{chaincode}.%{success}             | histogram | The time to build a chaincode image in seconds.            |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| endorser.chaincode_instantiation_failures.%{channel}.%{chaincode}                       | counter   | The number of chaincode instantiations or upgrade that     |
|                                                                                         |           | have failed.                                               |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| endorser.duplicate_transaction_failures.%{channel}.%{chaincode}                         | counter   | The number of failed proposals due to duplicate            |
|                                                                                         |           | transaction ID.                                            |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| endorser.endorsement_failures.%{channel}.%{chaincode}.%{chaincodeerror}                 | counter   | The number of failed endorsements.                         |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| endorser.proposal_acl_failures.%{channel}.%{chaincode}                                  | counter   | The number of proposals that failed ACL checks.            |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| endorser.proposal_duration.%{channel}.%{chaincode}.%{success}                           | histogram | The time to complete a proposal.                           |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| endorser.proposal_simulation_failures.%{channel}.%{chaincode}                           | counter   | The number of failed proposal simulations                  |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| endorser.proposal_validation_failures                                                   | counter   | The number of proposals that have failed initial           |
|                                                                                         |           | validation.                                                |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| endorser.proposals_received                                                             | counter   | The number of proposals received.                          |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| endorser.successful_proposals                                                           | counter   | The number of successful proposals.                        |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| fabric_version.%{version}                                                               | gauge     | The active version of Fabric.                              |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| gossip.comm.messages_received                                                           | counter   | Number of messages received                                |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| gossip.comm.messages_sent                                                               | counter   | Number of messages sent                                    |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| gossip.comm.overflow_count                                                              | counter   | Number of outgoing queue buffer overflows                  |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| gossip.leader_election.leader.%{channel}                                                | gauge     | Peer is leader (1) or follower (0)                         |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| gossip.membership.total_peers_known.%{channel}                                          | gauge     | Total known peers                                          |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| gossip.payload_buffer.size.%{channel}                                                   | gauge     | Size of the payload buffer                                 |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| gossip.privdata.commit_block_duration.%{channel}                                        | histogram | Time it takes to commit private data and the corresponding |
|                                                                                         |           | block (in seconds)                                         |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| gossip.privdata.fetch_duration.%{channel}                                               | histogram | Time it takes to fetch missing private data from peers (in |
|                                                                                         |           | seconds)                                                   |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| gossip.privdata.list_missing_duration.%{channel}                                        | histogram | Time it takes to list the missing private data (in         |
|                                                                                         |           | seconds)                                                   |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| gossip.privdata.pull_duration.%{channel}                                                | histogram | Time it takes to pull a missing private data element (in   |
|                                                                                         |           | seconds)                                                   |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| gossip.privdata.purge_duration.%{channel}                                               | histogram | Time it takes to purge private data (in seconds)           |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| gossip.privdata.reconciliation_duration.%{channel}                                      | histogram | Time it takes for reconciliation to complete (in seconds)  |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| gossip.privdata.retrieve_duration.%{channel}                                            | histogram | Time it takes to retrieve missing private data elements    |
|                                                                                         |           | from the ledger (in seconds)                               |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| gossip.privdata.send_duration.%{channel}                                                | histogram | Time it takes to send a missing private data element (in   |
|                                                                                         |           | seconds)                                                   |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| gossip.privdata.validation_duration.%{channel}                                          | histogram | Time it takes to validate a block (in seconds)             |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| gossip.state.commit_duration.%{channel}                                                 | histogram | Time it takes to commit a block in seconds                 |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| gossip.state.height.%{channel}                                                          | gauge     | Current ledger height                                      |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| grpc.comm.conn_closed                                                                   | counter   | gRPC connections closed. Open minus closed is the active   |
|                                                                                         |           | number of connections.                                     |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| grpc.comm.conn_opened                                                                   | counter   | gRPC connections opened. Open minus closed is the active   |
|                                                                                         |           | number of connections.                                     |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| grpc.server.stream_messages_received.%{service}.%{method}                               | counter   | The number of stream messages received.                    |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| grpc.server.stream_messages_sent.%{service}.%{method}                                   | counter   | The number of stream messages sent.                        |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| grpc.server.stream_request_duration.%{service}.%{method}.%{code}                        | histogram | The time to complete a stream request.                     |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| grpc.server.stream_requests_completed.%{service}.%{method}.%{code}                      | counter   | The number of stream requests completed.                   |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| grpc.server.stream_requests_received.%{service}.%{method}                               | counter   | The number of stream requests received.                    |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| grpc.server.unary_request_duration.%{service}.%{method}.%{code}                         | histogram | The time to complete a unary request.                      |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| grpc.server.unary_requests_completed.%{service}.%{method}.%{code}                       | counter   | The number of unary requests completed.                    |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| grpc.server.unary_requests_received.%{service}.%{method}                                | counter   | The number of unary requests received.                     |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| ledger.block_processing_time.%{channel}                                                 | histogram | Time taken in seconds for ledger block processing.         |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| ledger.blockchain_height.%{channel}                                                     | gauge     | Height of the chain in blocks.                             |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| ledger.blockstorage_and_pvtdata_commit_time.%{channel}                                  | histogram | Time taken in seconds for committing the block and private |
|                                                                                         |           | data to storage.                                           |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| ledger.blockstorage_commit_time.%{channel}                                              | histogram | Time taken in seconds for committing the block to storage. |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| ledger.statedb_commit_time.%{channel}                                                   | histogram | Time taken in seconds for committing block changes to      |
|                                                                                         |           | state db.                                                  |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| ledger.transaction_count.%{channel}.%{transaction_type}.%{chaincode}.%{validation_code} | counter   | Number of transactions processed.                          |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| logging.entries_checked.%{level}                                                        | counter   | Number of log entries checked against the active logging   |
|                                                                                         |           | level                                                      |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
| logging.entries_written.%{level}                                                        | counter   | Number of log entries that are written                     |
+-----------------------------------------------------------------------------------------+-----------+------------------------------------------------------------+
```

**Review by sykesm — CHANGES_REQUESTED** [review]
This tool is used in the CA as well. That means having Orderer and Peer metrics don't make sense in that context and, when we try to bump the metrics package there, we'll have to scratch our head and try to normalize what we have there...

Can we make these changes in a way that is backwards compatible with what exists today?

So

**Comment by pamandrejko** [comment]
I ran this through the doc build and found the Labels column unclear. 
Screen Shot 2020-02-05 at 4 22 28 PM

Just wanted to confirm this is the intended formatting and maybe we need to explain the purpose of the Labels column?

I can share the formatted .html file if you want to see it, but github won't let me attach it here.

**Comment by stephyee** [comment]
> I ran this through the doc build and found the Labels column unclear.
> Screen Shot 2020-02-05 at 4 22 28 PM
> 
> Just wanted to confirm this is the intended formatting and maybe we need to explain the purpose of the Labels column?
> 
> I can share the formatted .html file if you want to see it, but github won't let me attach it here.

@pamandrejko Definitely not intended formatting - we will take a closer look.

**Comment by sykesm** [comment]
See #651 for script based approach. We should do a follow up to had explicit label help to metrics to fill out information in the metric doc. The current doc contains the labels but the code has not be updated with help documentation.

**Pull request closed without merging**